diff --git a/ckan/controllers/user.py b/ckan/controllers/user.py
index d50ae61..d8f1108 100644
--- a/ckan/controllers/user.py
+++ b/ckan/controllers/user.py
@@ -432,6 +432,8 @@ class UserController(base.BaseController):
                         # and user_list does not return them
                         del data_dict['q']
                         data_dict['id'] = user_list[0]['id']
+                        # Get rid of auth audit on the context otherwise we'll get an exception
+                        context.pop('__auth_audit', None)
                         user_dict = get_action('user_show')(context, data_dict)
                         user_obj = context['user_obj']
                     elif len(user_list) > 1:
@@ -443,10 +445,13 @@ class UserController(base.BaseController):
 
             if user_obj:
                 try:
-                    mailer.send_reset_link(user_obj)
-                    h.flash_success(_('Please check your inbox for '
-                                    'a reset code.'))
-                    h.redirect_to('/')
+                    if user_obj.state == 'active':
+                        mailer.send_reset_link(user_obj)
+                        h.flash_success(_('Please check your inbox for '
+                                        'a reset code.'))
+                        h.redirect_to('/')
+                    else:
+                        h.flash_error(_('Could not send reset link as account is not yet approved/active, please contact the system administrator'))
                 except mailer.MailerException, e:
                     h.flash_error(_('Could not send reset link: %s') %
                                   unicode(e))

